### PR TITLE
feat(thermocycler-gen2): thermal calibration and tuning

### PR DESCRIPTION
**Pull request by fsinapi**
### Summary

- Added in a new calibration offset for the thermal system which takes the current heat sink temperature into account when calculating the offset from thermistor to plate, copied from the Thermocycler Gen1.
- Modified default PID values for peltiers to be less aggressive in order to reduce oscillation
- Added default thermal calibration values based off of initial population testing on EVT units

Tested on an EVT unit. The offset values were verified by testing the steady-state temperature in the well plates using a multiprobe tool.

### CODE REVIEW

File: stm32-modules/include/thermocycler-gen2/thermocycler-gen2/eeprom.hpp [hunk 1]
```diff
@@ -23,7 +23,7 @@ namespace eeprom {
  * readings. Using two constants, B and C (for legacy purposes), the
```
Comment (sfoster1) [inline stm32-modules/include/thermocycler-gen2/thermocycler-gen2/eeprom.hpp:23]: add an explanation about A?

File: stm32-modules/include/thermocycler-gen2/thermocycler-gen2/eeprom.hpp [hunk 2]
```diff
@@ -118,12 +130,17 @@ class Eeprom {
...
+        // See \ref EEPROMFlag
+        CONST_FLAG = 2,
+        CONST_A = 3,  // Value of the A constant
     };
 
     // Enumeration of the EEPROM_CONST_FLAG values
-    enum class EEPROMFlag { WRITTEN_NO_CHECKSUM = 1, INVALID = 0xFF };
+    enum class EEPROMFlag {
+        WRITTEN_NO_A = 1,    // Values of B and C are written
+        WRITTEN_WITH_A = 2,  // Values of all constants are written
```
Comment (sfoster1) [inline stm32-modules/include/thermocycler-gen2/thermocycler-gen2/eeprom.hpp]: you are a hero for trying to keep this backward compatible across proto versions but maybe we should also consider just considering eeproms without A invalid?

File: stm32-modules/include/thermocycler-gen2/thermocycler-gen2/thermal_plate_task.hpp [hunk 3]
```diff
@@ -661,7 +672,9 @@ class ThermalPlateTask {
 
     auto handle_temperature_conversion(uint16_t conversion_result,
                                        Thermistor& thermistor,
-                                       bool apply_offset) -> void {
+                                       bool apply_offset,
+                                       const double heatsink_temp = 0.0F)
```
Comment (sfoster1) [inline stm32-modules/include/thermocycler-gen2/thermocycler-gen2/thermal_plate_task.hpp]: [nit] I personally don't love const value params

**Review by sfoster1 — APPROVED** [review]
Looks good but we're gonna want to update some comments.

File: stm32-modules/include/thermocycler-gen2/thermocycler-gen2/thermal_plate_task.hpp (hunk 3 above)
Comment (fsinapi) [reply stm32-modules/include/thermocycler-gen2/thermocycler-gen2/thermal_plate_task.hpp]: image

**Pull request merged**